File: cpp/base/src/main/vsdk/toolkit/media/RGBProceduralColorPalette.cpp
#include "vsdk/toolkit/media/RGBProceduralColorPalette.h"
#include "vsdk/toolkit/common/color/ColorRgb.h"

RGBProceduralColorPalette::RGBProceduralColorPalette() : RGBColorPalette() {
    pure = true;
}

RGBProceduralColorPalette::~RGBProceduralColorPalette() {
}

int RGBProceduralColorPalette::selectNearestIndexToRgb(const ColorRgb& c) const {
    return RGBColorPalette::selectNearestIndexToRgb(c);
}

void RGBProceduralColorPalette::setColorAt(int i, ColorRgb* c) {
    pure = false;
    RGBColorPalette::setColorAt(i, c);
}

void RGBProceduralColorPalette::setColorAt(int i, double r, double g, double b) {
    pure = false;
    RGBColorPalette::setColorAt(i, r, g, b);
}

void RGBProceduralColorPalette::addColor(ColorRgb* c) {
    pure = false;
    RGBColorPalette::addColor(c);
}

void RGBProceduralColorPalette::addColor(double r, double g, double b) {
    pure = false;
    RGBColorPalette::addColor(r, g, b);
}

File: typescript/base/src/main/vsdk/toolkit/media/RGBProceduralColorPalette.ts
import { RGBColorPalette } from "./RGBColorPalette.js";
import { ColorRgb } from "../common/color/ColorRgb.js";
export abstract class RGBProceduralColorPalette extends RGBColorPalette {
    protected pure = true;
    public override selectNearestIndexToRgb(color: ColorRgb): number {
        return super.selectNearestIndexToRgb(color);
    }
    public override setColorAt(i: number, c: ColorRgb): void;
    public override setColorAt(i: number, r: number, g: number, b: number): void;
    public override setColorAt(i: number, a: ColorRgb | number, b?: number, c?: number): void {
        this.pure = false;
        if (a instanceof ColorRgb) super.setColorAt(i, a);
        else super.setColorAt(i, a, b!, c!);
    }
    public override addColor(c: ColorRgb): void;
    public override addColor(r: number, g: number, b: number): void;
    public override addColor(a: ColorRgb | number, b?: number, c?: number): void {
        this.pure = false;
        if (a instanceof ColorRgb) super.addColor(a);
        else super.addColor(a, b!, c!);
    }
}
